eva/parser/lark_parser.py

```diff
@@ -0,0 +1,58 @@
+# coding=utf-8
+# Copyright 2018-2022 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+
+from lark import Lark
+
+from eva.parser.lark_visitor import LarkInterpreter
+
+
+class LarkParser(object):
+    """
+    Parser for EVA QL based on Lark
+    """
+
+    _instance = None
+    _parser = None
+
+    def __new__(cls):
```

Comment:
With this singleton implementation, `__init__` gets called multiple times. https://stackoverflow.com/questions/12305142/issue-with-singleton-python-call-two-times-init

Reply:
Done